data-structures/queues-1.js

```diff
@@ -0,0 +1,16 @@
+export function getFront(queue) {
+  return queue.peek() === undefined ? undefined : queue.peek();
+}
+
+export function addToBack(queue, value) {
+  queue.enqueue(value);
+}
+
+export function takeFront(queue) {
+  return queue.peek() === undefined ? undefined : queue.dequeue();
+}
+
+export function isEmpty(queue) {
+  const fontValue = queue.peek();
+  return typeof fontValue === 'undefined';
```

Comment:
Checking `typeof` for `'undefined'` is safe but old fashioned. In modern JS there's no reason to not just do `return fontValue === undefined;`.